personalities/sctfe/storage.go

```diff
@@ -102,43 +101,26 @@ type cachedIssuerStorage struct {
 	s IssuerStorage
 }
 
-// Exists checks if the key exists in the local cache, if not checks in the underlying storage.
-// If it finds it there, caches the key locally.
-func (c *cachedIssuerStorage) Exists(ctx context.Context, key []byte) (bool, error) {
-	c.RLock()
-	_, ok := c.m[string(key)]
-	c.RUnlock()
-	if ok {
-		klog.V(2).Infof("Exists: found %q in local key cache", key)
-		return true, nil
-	}
-	ok, err := c.s.Exists(ctx, key)
-	if err != nil {
-		return false, fmt.Errorf("error checking if issuer %q exists in the underlying IssuerStorage: %s", key, err)
-	}
-	if ok {
-		c.Lock()
-		c.m[string(key)] = struct{}{}
-		c.Unlock()
-	}
-	return ok, nil
+// NewCachedIssuerStorage returns a new local cache to wrapping an IssuerStorage
+func NewCachedIssuerStorage(s IssuerStorage) *cachedIssuerStorage {
```

Comment:
You _could_ just return a func() here if the only place the return value is used is on L65 to grab a reference to the `Add...` method.

Reply:
I went even further and removed the cachedIssuerStorage object.